### ISSUE TITLE
(False positive) Method expected to be 'property', found it instead as 'method'.

### ISSUE THREAD

**Issue by staykotzenov** [PyCQA/pylint#4023]
### Steps to reproduce
1. Creating a base class with a `@property` decorator on one of it`s field methods.
2. Inheriting the base class in a child class and overriding the property method, this time with a `@cached_property` decorator.

```
# test.py
class Test():

    def __init__(self, smth):
        self.smth=smth

    @property
    def test_property(self):
        return 5
```
```
import functools
from test import Test

#child.py
class Child(Test):

    @functools.cached_property
    def test_property(self):
        return super().test_property + 5

```
Please refer to image below

### Current behavior
Error message is triggered. 
`Method 'test_property' was expected to be 'property', found it instead as 'method'`

### Expected behavior

No error should be displayed, since `@cached_property` should still be a property

### pylint --version output
pilyint 2.5.2
astroid 1.4.1
Python 3.8.0 (default, Nov 23 2019, 05:49:00)
[GCC 8.3.0]

**Comment by hippo91** [comment]
@staykotzenov thanks for the report. I can reproduce it with this smaller snippet:
```
import functools
class Test():
    @property
    def test_property(self):
        return 5


class Child(Test):
    @functools.cached_property
    def test_property(self):
        return super().test_property + 5
```
Of course if the `test_property` of the child class is decorated with `@property` then the message is not emitted.

**Comment by hippo91** [comment]
To those willing to fix this issue, it seems that the test `inferred.name == 'property'` inside the `_is_property_decorator ` function of the `checkers/utils.py` is probably to restrictive.
The use of the `_is_property` function of the astroid's `bases.py` module is probably more adapted here.
